sdk/servicebus/service-bus/src/track2/serviceBusReceiverClient.ts

```diff
@@ -0,0 +1,568 @@
+   *
+   * @param queueAuth Data needed to connect to a queue.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (queueAuth: QueueAuth, receiveMode: "peekLock", session: Session): ClientTypeT<
+    "peekLock",
+    "queue",
+    "sessions"
+  >;
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param queueAuth Data needed to connect to a queue.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (queueAuth: QueueAuth, receiveMode: "receiveAndDelete", session: Session): ClientTypeT<
+    "receiveAndDelete",
+    "queue",
+    "sessions"
+  >;
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param subscriptionAuth Data needed to connect to a subscription.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (subscriptionAuth: SubscriptionAuth, receiveMode: "peekLock"): ClientTypeT<
+    "peekLock",
+    "subscription",
+    "nosessions"
+  >;
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param subscriptionAuth Data needed to connect to a subscription.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (subscriptionAuth: SubscriptionAuth, receiveMode: "receiveAndDelete"): ClientTypeT<
+    "receiveAndDelete",
+    "subscription",
+    "nosessions"
+  >;
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param subscriptionAuth Data needed to connect to a subscription.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (subscriptionAuth: SubscriptionAuth, receiveMode: "peekLock", session: Session): ClientTypeT<
+    "peekLock",
+    "subscription",
+    "sessions"
+  >;
+
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param subscriptionAuth Data needed to connect to a subscription.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
+   * @param options Options for the client itself.
+   */
+  new (
+    subscriptionAuth: SubscriptionAuth,
+    receiveMode: "receiveAndDelete",
+    session: Session
+  ): ClientTypeT<"receiveAndDelete", "subscription", "sessions">;
+}
+
+/**
+ * Implementation class for receivers.
+ * @internal
+ * @ignore
+ */
+export class ReceiverClientImplementation {
+  constructor(
+    auth1: QueueAuth | SubscriptionAuth,
+    receiveMode2: "peekLock" | "receiveAndDelete",
+    sessionOrOptions3?: Session | ServiceBusClientOptions,
+    options4?: ServiceBusClientOptions
+  ) {
+    let options: ServiceBusClientOptions;
+    let session: Session | undefined;
+
+    if (sessionOrOptions3 != null && isSession(sessionOrOptions3)) {
```

Comment:
Nit: I think `isSession` should tell you true or false even if the input is null:
```suggestion
    if (isSession(sessionOrOptions3)) {
```

Reply:
You're 100% right about that. Removed this check and fixed isSession()